### PR TITLE
Add telemetry to HTTP requests

### PR DESCRIPTION
**Pull request by joshcanhelp**
### Description

Add a telemetry header to HTTP requests.

### Testing

- [x] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [x] All active GitHub checks for tests, formatting, and security are passing

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -4,9 +4,18 @@ const url = require('url');
 const urlJoin = require('url-join');
 const pkg = require('../package.json');
 
+const telemetryHeader = {
+  name: 'express-oidc',
```
Comment (joshcanhelp) [inline lib/client.js]: Shortened the name a bit but still recognizable.

File: lib/client.js [hunk 2]
```diff
@@ -4,9 +4,18 @@ const url = require('url');
...
+  version: pkg.version,
+  env: {
+    node: process.version
+  }
+};
+
 custom.setHttpOptionsDefaults({
   headers: {
-    'User-Agent': `${pkg.name}/${pkg.version} (${pkg.homepage})`
+    'User-Agent': `${pkg.name}/${pkg.version}`,
```
Comment (joshcanhelp) [inline lib/client.js]: Is this a good user agent? I removed the homepage, seems like unnecessary data.

File: test/client.tests.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      clientSecret: '__test_client_secret__',
+      issuerBaseURL: 'https://flosser.auth0.com',
+      baseURL: 'https://theapplication.com',
+    });
+
+    let client;
+    before(async function() {
+      client = await getClient(config);
+
+      nock('https://flosser.auth0.com')
```
Comment (joshcanhelp) [inline test/client.tests.js]: Nock was the best way I could think of/find for getting the options being used. They are not (understandably) exposed outside of the library.

File: lib/client.js (hunk 2 above)
Comment (stevehobbsdev) [reply lib/client.js]: @joshcanhelp Do you care about Node version or possibly the OS?

If not, package name and version is probably enough.

Comment (joshcanhelp) [reply lib/client.js]: Node version is sent in the `env` object. We don't parse the user agent (at this point) for telemetry.

**Pull request merged**